### DISCUSSION TITLE
[Question][Hw5] memory in Analysis.check_instr

### DISCUSSION THREAD

**Discussion by nubi12336** [prosyslab-classroom/cs348-information-security#630] [Q&A]
In example1.c, there is `if (x<0)`  and it is `%div3 = sdiv i32 10, %call, !dbg !23` in example1.ll, `if.then2`.
I printed out a table and I got a result `%div3 = sdiv i32 10, %call, !dbg !23 -> %call -> Neg`.
Then, I thought, in Analysis.check_instr, as it matches with Llvm.Opcode.SDiv and `v` is `Neg`(as `e` is `%call`)
`Memory.Value.order zero v` must be false.
However, it prints out `Potential Division-by-zero @ example1.c:main:9:14, %call = Top`.

What is wrong?

**Comment by nubi12336** [comment]
I printed out in `Memory.pp Format.std_formatter`  memory Analysis.check_instr.
Below is result.
![image](https://user-images.githubusercontent.com/79691513/120648261-64d98100-c4b6-11eb-87d7-1dcf69062bff.png)
I think second row is weird..

**Comment by RiceBiscuits** [comment]
I can't find the bug due to the lack of information.
However, I guess `analysis.run` function or` transfer_* `may have a problem. 
Please debug your` run` and `transfer_* `function carefully.

**Comment by nubi12336** [comment]
How come it can happen that, in the resulting table there is `%div3 = sdiv i32 10, %call, !dbg !23 -> %call -> Neg`, but in the result div3, **%call become top**?(above picture)
%div3 ~ is 10/x in `if (x < 0) {z = 10 / x}`.
